FWCore/Services/plugins/Timing.cc

```diff
@@ -310,6 +316,42 @@ namespace edm {
         iRegistry.watchPostSourceConstruction(this, &Timing::postModule);
       }
 
+      iRegistry.preESModuleSignal_.connect([this](auto const& key, auto const& context) {
+        //find available slot
+        auto startTime = getTime();
+        bool foundSlot = false;
+        do {
+          for (size_t i = 0; i < eventSetupModuleStartTimes_.size(); ++i) {
+            auto& slot = *eventSetupModuleStartTimes_[i];
+            double expect = 0.;
+            if (slot.compare_exchange_strong(expect, startTime)) {
+              foundSlot = true;
+              eventSetupModuleCallInfo_[i].first = uintptr_t(context.componentDescription());
+              eventSetupModuleCallInfo_[i].second = key;
+              break;
+            }
+          }
+          //if foundSlot == false then other threads stole the slots before this thread
+          // so should check starting over again
+        } while (not foundSlot);
+      });
+      iRegistry.postESModuleSignal_.connect([this](auto const& key, auto const& context) {
+        auto stopTime = getTime();
+        for (size_t i = 0; i < eventSetupModuleStartTimes_.size(); ++i) {
+          auto const& info = eventSetupModuleCallInfo_[i];
+          if (info.first == uintptr_t(context.componentDescription()) and info.second == key) {
+            auto startTime = eventSetupModuleStartTimes_[i]->load();
+            eventSetupModuleStartTimes_[i]->store(0.);
```

Comment:
Would
```suggestion
            auto startTime = eventSetupModuleStartTimes_[i]->exchange(0.);
```
make sense?